### DISCUSSION TITLE
Extract `cnrc` into separate project

### DISCUSSION THREAD

**Discussion by tzdybal** [rollkit/rollkit#409] [Ideas]
https://github.com/celestiaorg/optimint/pull/399 introduces simple library to handle celestia-node RPC endpoints.

In my opinion, this library can be useful on it's own. My idea is:
 1. extract this into separate project (name proposals are welcome!)
 1. prepare testing environment (testing unit tests with ephemeral-cluster)
 1. invite community to implement missing endpoints - create issues for all missing features

**Comment by adlerjohn** [comment]
Not a bad idea. Would also make it easier to implement in different languages.

**Comment by jbowen93** [comment]
I definitely think it should be a separate library. `celestia-node-client`.

It's important to note this isn't a RPC client. This is a REST client.

**Comment by Wondertan** [comment]
I like the idea of extracting alpha API into a separate repo, so we can easily deprecate it once the time comes. Surely, there are also other good reasons to extract it. 

Note that future Public API in celestia-node inherently covers the client-side of the communication. Ideally, Public API would be a set of interfaces over each celestia node module with two implementations: one over an embedded node and another over an RPC client that talks to a remote Node. The latter will supersede the Alpha API eventually, and these interfaces would live in the celestia node repo or extract to a separate API repo, which is TBD.
